core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -375,7 +376,16 @@ private static TableMetadata create(TableOperations ops, UpdateTableRequest requ
     request.requirements().forEach(requirement -> requirement.validate(ops.current()));
 
     TableMetadata.Builder builder = TableMetadata.buildFromEmpty();
-    request.updates().forEach(update -> update.applyTo(builder));
+    request
+        .updates()
+        .forEach(
+            update -> {
+              if (update instanceof UpgradeFormatVersion) {
+                builder.setInitialFormatVersion(((UpgradeFormatVersion) update).formatVersion());
```

Comment:
I don't think this is the right way to set the format version. The initial metadata needs to be created as v1, rather than dangerously setting the version to 1 after constructing it.

Reply:
@rdblue Thanks for your comment. Do you mean that we should expose something like `buildFromSpecifiedVersion(formatVersion)` in `TableMetadata` for getting a builder with specified format version, and check the updates list in advance here to get the format version before creating the builder?

Reply:
Or should we use a builder that lazily initialize `formatVersion` as @nastra described above?

Reply:
Or an alternative is to add a `buildFromEmpty(int formatVersion)` API and apply the updates. Either way, agree with the idea that the format version should be defined once upfront when building a new table metadata instance rather than mutate it later.

Reply:
@hantangwangd I've raised a PR to your branch https://github.com/hantangwangd/iceberg/pull/1/files to show what I mean. cc @nastra @rdblue @aokolnychyi

Reply:
@amogh-jahagirdar Thank you for the change, that makes sense. I absorbed it into this PR.